src/django_mysql/models/lookups.py

```diff
@@ -101,9 +101,29 @@ def as_sql(self, qn, connection):
         return "JSON_CONTAINS({}, {})".format(rhs, lhs), params
 
 
-class JSONContains(JSONLookupMixin, Lookup):
+class JSONContains(Lookup):
     lookup_name = "contains"
 
+    if django.VERSION >= (3, 0):
+
+        def get_prep_lookup(self):
+            value = self.rhs
+            if hasattr(value, "resolve_expression"):
+                raise ValueError(
+                    "JSONField's 'contains' lookup only works with literal values"
+                )
+            return JSONValue(value)
+
+    else:
+
+        def get_prep_lookup(self):
+            value = self.rhs
+            if hasattr(value, "_prepare"):
+                raise ValueError(
+                    "JSONField's 'contains' lookup only works with literal values"
+                )
+            return JSONValue(value)
```

Comment:
Why block expressions, rather than casting their results? This is backwards incompatible.

Reply:
Expressions never worked with the contains lookup.

`JSONModel.objects.filter(attrs__a__contains=F("name"))`  was translated into
```mysql
SELECT ...
FROM `testapp_jsonmodel`
WHERE JSON_CONTAINS(JSON_EXTRACT(`testapp_jsonmodel`.`attrs`, $.a), (`testapp_jsonmodel`.`name`))
```
and raised:
`django.db.utils.OperationalError: (3141, 'Invalid JSON text in argument 2 to function json_contains: "Invalid value." at position 0.')`

Maybe there's a way to make it work, but this PR is not backwards incompatible.

Reply:
Okay cool, we're just raising a more sensible error.